### PR TITLE
Adding test to make sure pending dynamic link is getting passed to App delegate method.

### PR DESCRIPTION
**Pull request by eldhosembabu**
Adding unit test to make sure pending dynamic link is getting passed to proper App delegate method.

### CODE REVIEW

File: FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m [hunk 1]
```diff
@@ -1131,13 +1133,36 @@ - (void)test_retrievePendingDeepLinkShouldSetkFIRDLOpenURLKeyRegardlessOfFailure
...
   XCTAssertEqual([[NSUserDefaults standardUserDefaults] boolForKey:kFIRDLOpenURLKey], YES,
                  @"kFIRDLOpenURL key should be set regardless of failures");
 }
 
+- (void)test_passRetrievedDynamicLinkToApplicationDelegatesProperly {
+  // Creating ApplicationDelegate partial mock object.
+  id applicationDelegate = OCMPartialMock([UIApplication sharedApplication].delegate);
+  // Creating FIRDynamicLinks partial mock object.
+  id firebaseDynamicLinks = OCMPartialMock(self.service);
+  // Stubing Application delegate to return YES when application:openURL:options method is called.
```
Comment (paulb777) [inline FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m:1148]: nit: Stubbing

File: FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m [hunk 2]
```diff
@@ -1131,13 +1133,36 @@ - (void)test_retrievePendingDeepLinkShouldSetkFIRDLOpenURLKeyRegardlessOfFailure
...
+  // Not sure why this is required as we are not concerned about its return, but without this, the
+  // test will throw NSInvalidArgumentException with message "unrecognized selector sent to
+  // instance".
+  OCMStub([applicationDelegate application:[OCMArg any] openURL:[OCMArg any] options:[OCMArg any]])
+      .andReturn(YES);
+  // Stubing firebase dynamiclinks instance to return YES when isOpenUrlMethodPresentInAppDelegate
```
Comment (paulb777) [inline FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m:1154]: Stubbing

**Review by paulb777 — APPROVED** [review]
Thanks for adding the test!

File: FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m (hunk 1 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m]: done

File: FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m (hunk 2 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m]: done

**Comment by christibbs** [comment]
Love the workaround!

**Review by christibbs — APPROVED** [review]

**Pull request merged**